### PR TITLE
fix: add check for SSR on Picasso root guard

### PR DESCRIPTION
**Pull request by augustobmoura**
[FX-3873]

### Description

Some time ago we added a guard to Picasso root so it only render children after the root itself is mounted on the DOM, this is because when loading a page for the first time on a modal/popup, it was attached to the body of the page instead of the proper location of Picasso root. This guard doesn't make sense on SSR, where no component is truly "mounted"

### How to test

- Tests should pass
- On a project with SSR enabled the first render should respond with a proper static HTML ready for hydratation

### Screenshots

No screenshots

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [NA] Annotate all `props` in component with documentation
- [NA] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests

**Breaking change**

- [x] codemod is created and showcased in the changeset
- [x] test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-3873]: https://toptal-core.atlassian.net/browse/FX-3873?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by augustobmoura** [comment]
@toptal-bot run package:alpha-release

**Comment by augustobmoura** [comment]
@toptal-anvil ping reviewers

File: packages/picasso-provider/src/Picasso/PicassoGlobalStylesProvider.tsx [hunk 1]
```diff
@@ -15,6 +15,8 @@ export interface PicassoGlobalStylesProviderProps extends TextLabelProps {
   disableTransitions?: boolean
 }
 
+const isBrowser = typeof window !== 'undefined' && !!window.document
```
Comment (dmaklygin) [inline packages/picasso-provider/src/Picasso/PicassoGlobalStylesProvider.tsx:18]: we have it in `picasso-shared`

```
import { isBrowser } from '@toptal/picasso-shared'
```

**Review by dmaklygin — APPROVED** [review]

**Review by mkrl — APPROVED** [review]

File: packages/picasso-provider/src/Picasso/PicassoGlobalStylesProvider.tsx (hunk 1 above)
Comment (augustobmoura) [reply packages/picasso-provider/src/Picasso/PicassoGlobalStylesProvider.tsx]: Done :+1:

**Pull request merged**